src/gap.c

```diff
@@ -1649,9 +1648,10 @@ void InitializeGap (
         /* Call POST_RESTORE which is a GAP function that now takes control, 
            calls the post restore functions and then runs a GAP session */
         if (POST_RESTORE != 0 && IS_FUNC(POST_RESTORE)) {
-          TRY_IF_NO_ERROR {
+          GAP_TRY {
             CALL_0ARGS(POST_RESTORE);
           }
+          GAP_CATCH {}
```

Comment:
This change is correct because in case of a failure, it doesn't matter whether we `NrError` to a non-zero value or: before we enter a REPL, we'll call `ClearError` anyway.